[Mario Alvaro] iP

marioalvaro opened the pull request:
### DukePro

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/nus-cs2103-AY2324S1/ip/pull/129).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [ ] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

marioalvaro committed:
level 0

marioalvaro committed:
level 1

marioalvaro committed:
level 2

marioalvaro committed:
level 3

marioalvaro committed:
level 4

marioalvaro committed:
testing

marioalvaro committed:
level-5

marioalvaro committed:
level-6

marioalvaro committed:
Level-7

marioalvaro committed:
Merge pull request #1 from marioalvaro/branch-Level-7

Level-7

marioalvaro committed:
Level-8

marioalvaro committed:
Merge pull request #2 from marioalvaro/branch-Level-8

Level-8

marioalvaro committed:
A-MoreOOP

marioalvaro committed:
Merge pull request #3 from marioalvaro/branch-A-MoreOOP

A-MoreOOP

marioalvaro committed:
A-Packages

marioalvaro committed:
Merge pull request #4 from marioalvaro/branch-A-Packages

A-Packages

marioalvaro committed:
Merge pull request #5 from marioalvaro/add-gradle-support

Add gradle support

marioalvaro committed:
A-Gradle

marioalvaro committed:
A-JUnit

marioalvaro committed:
Merge pull request #6 from marioalvaro/branch-A-JUnit

A-JUnit

marioalvaro committed:
A-Jar

marioalvaro committed:
Merge pull request #7 from marioalvaro/branch-A-Jar

A-Jar

marioalvaro committed:
Delete data directory

marioalvaro committed:
Merge pull request #8 from marioalvaro/marioalvaro-fix-delete-data-directory

Delete data directory

marioalvaro committed:
Level-9

marioalvaro committed:
Add JavaDoc

marioalvaro committed:
A-CodingStandard

marioalvaro committed:
Merge pull request #9 from marioalvaro/branch-A-CodingStandard

A-CodingStandard

marioalvaro committed:
Merge pull request #10 from marioalvaro/branch-Level-9

Level-9

marioalvaro committed:
Merge branch 'master' into branch-A-JavaDoc

marioalvaro committed:
Merge pull request #11 from marioalvaro/branch-A-JavaDoc

Add JavaDoc

TiwKangXu commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,65 @@
+
+import java.util.Scanner;
+
 public class Duke {
```
Wouldn't Checkstyle give error for not having header comments for Duke class? 🤔

TiwKangXu commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,65 @@
+import duke.task.TaskList;
+import duke.ui.Ui;
+
+import java.util.Scanner;
```
Should java.util.Scanner be imported before the packages?

TiwKangXu commented on src/main/java/duke/command/ByeCommand.java:
```diff
@@ -0,0 +1,26 @@
+package duke.command;
```
I like the way you handle actions using independent Classes.

TiwKangXu commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,79 @@
+        try {
+            CommandWord commandWord = CommandWord.valueOf(input.split(" ")[0].toUpperCase());
+
+            String[] splitTask = input.split(" ", 2);
```
I like the way you break a string into array of strings and use it among classes.

TiwKangXu commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,79 @@
+                    if (splitTask.length == 2) {
+                        return new UnmarkCommand(splitTask);
+                    }
+                case TODO:
+                case EVENT:
+                case DEADLINE:
```
Interesting approach. Does this work? I have not tried this before.

TiwKangXu commented on src/main/java/duke/task/Todo.java:
```diff
@@ -0,0 +1,39 @@
+    /**
+     * the type Icon
+     */
+    private final String TYPE = "T";
```
I like the way you set tasks' label as a final.

TiwKangXu reviewed (commented):
LGTM. Neatly written and enlighting code.

marioalvaro committed:
Add CheckStyle

marioalvaro committed:
Merge pull request #12 from marioalvaro/branch-A-CheckStyle

Add CheckStyle

marioalvaro committed:
Add level 10

marioalvaro committed:
Merge pull request #13 from marioalvaro/branch-Level-10

Add level 10

Cloud7050 commented on src/main/java/DialogBox.java:
```diff
@@ -0,0 +1,80 @@
+     * Flips the dialog box such that the ImageView is on the left and text on the right.
+     */
+    private void flip() {
+        ObservableList<Node> tmp = FXCollections.observableArrayList(this.getChildren());
+        Collections.reverse(tmp);
+        getChildren().setAll(tmp);
```
Perhaps there is a better name for this `tmp` variable?

![image](https://github.com/nus-cs2103-AY2324S1/ip/assets/44526554/32ae0e12-f4bf-4fae-8a20-f4d72e14ba77)

Cloud7050 commented on src/main/java/DialogBox.java:
```diff
@@ -0,0 +1,80 @@
+    /**
+     * Constructor for a DialogBox.
+     *
+     * @param text the text contained in the dialog.
+     * @param img The image of the user or Duke.
```
In some of the Javadocs for params, sometimes they start with a capital letter while other times a lowercase letter. Eg "the" and "The". Maybe it's worth making consistent?

For reference, the coding standard's example starts with a lowercase letter unless the word is something like a class name.

> ```java
> /**
>  * Returns lateral location of the specified position.
>  * If the position is unset, NaN is returned.
>  *
>  * @param x X coordinate of position.
>  * @param y Y coordinate of position.
>  * @param zone Zone of position.
>  * @return Lateral location.
>  * @throws IllegalArgumentException  If zone is <= 0.
>  */
> public double computeLocation(double x, double y, int zone)
>         throws IllegalArgumentException {
>     //...
> }
> ```

https://se-education.org/guides/conventions/java/intermediate.html#comments

Cloud7050 commented on src/main/java/DialogBox.java:
```diff
@@ -0,0 +1,80 @@
+     * @return A DialogBox with the information.
+     */
+    public static DialogBox getDukeDialog(String text, Image img) {
+        var db = new DialogBox(text, img);
```
Is `var` the best choice of type in this situation?

Cloud7050 commented on src/main/java/MainWindow.java:
```diff
@@ -0,0 +1,67 @@
+    }
+
+    /**
+     * set duke object.
+     * @param d input duke
```
I think there should be a line between the description and `@param`. Also, it may be worth taking a look at the various Javadocs' capitalisation.

![image](https://github.com/nus-cs2103-AY2324S1/ip/assets/44526554/73f3329e-ffe6-4fe9-ba4e-3e8db5ea88c7)

Cloud7050 commented on src/main/java/MainWindow.java:
```diff
@@ -0,0 +1,67 @@
+    private void handleUserInput() throws InterruptedException {
+        String input = userInput.getText();
+        String response = duke.getResponse(input);
+        dialogContainer.getChildren().addAll(
+                DialogBox.getUserDialog(input, userImage),
+                DialogBox.getDukeDialog(response, dukeImage)
+        );
```
I think this is just one level of nesting as opposed to a wrapped line. Therefore the indentation level would be the normal 4 spaces as the 8-space rule does not apply here.

Cloud7050 commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,23 @@
+import duke.task.TaskList;
+import duke.ui.Ui;
+
+/**
+ * Abstract class for the Command class
+ *
+ * @author marioalvaro
+ */
+public abstract class Command {
+    /**
+     * Method to execute the command
+     *
+     * @param taskList The used TaskList
+     * @param ui The ui object
+     * @param storage The storage used
+     * @return Message response from running the command.
+     * @throws Exception Throw Exception if needed
+     */
```
Is it worth doing a pass over some of the documentation for consistency? For example wrt fullstops & capitalisation.

Cloud7050 commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,47 @@
+     */
+    public FindCommand(String[] splitTask) {
+        this.splitTask = splitTask;
+
```
Perhaps this newline is not needed, but this is pretty nitpicky so really it's up to your preference.

Cloud7050 commented on src/main/java/duke/command/ListCommand.java:
```diff
@@ -0,0 +1,26 @@
+     */
+    @Override
+    public String execute(TaskList taskList, Ui ui, Storage storage) {
+
```
Continuing from the above, here is another newline that stood out to me.

Cloud7050 commented on src/main/java/duke/command/UnmarkCommand.java:
```diff
@@ -0,0 +1,55 @@
+ *
+ * @author marioalvaro
+ */
+public class UnmarkCommand extends Command {
```
I do like how every `Command` is split into its own file. It allows neatly scaling up to support a high number of commands, and can isolate potentially complex commands.

Cloud7050 commented on src/main/java/duke/exception/DukeException.java:
```diff
@@ -0,0 +1,18 @@
+     *
+     * @param e The Exception message
+     */
+    public DukeException(String e) {
+        super(e);
+    }
```
This looks to me like a default constructor that Java would be able to generate on its own, but I could be mistaken.

Cloud7050 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,84 @@
+            CommandWord commandWord = CommandWord.valueOf(input.split(" ")[0].toUpperCase());
+
+            String[] splitTask = input.split(" ", 2);
+            switch (commandWord) {
+            case BYE:
```
Nice job following the coding standard for switch-case.

Cloud7050 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,123 @@
+            String scanned = sc.nextLine();
+            String[] singleTaskArray = scanned.split(" / ");
+
+            //do a checking <------
```
I'm not entirely sure what this comment means. Perhaps just a `//TODO` for yourself? I understand that a lot of this is still WIP, for example I noticed some kind of template above like "Replace this stub with your completed method.". But just thought I'd mention this anyway.

![image](https://github.com/nus-cs2103-AY2324S1/ip/assets/44526554/c2a0aa46-4da0-4b73-8eb9-ec269a27e908)

Cloud7050 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,54 @@
+     */
+    @Override
+    public String toString() {
+        return "[" + TYPE + "]" + super.toString()
+                + " (by: " + Time.toString(this.time) + ")";
```
To avoid concatenation spam, I personally prefer to use `String.format()`. Maybe it is worth considering.

marioalvaro committed:
Add assertions #1

marioalvaro committed:
Add better Code Quality

marioalvaro committed:
Assertions final

marioalvaro committed:
Merge pull request #14 from marioalvaro/branch-A-Assertions

Add assertions

marioalvaro committed:
Merge branch 'master' into branch-A-CodeQuality

marioalvaro committed:
Merge pull request #15 from marioalvaro/branch-A-CodeQuality

Add better Code Quality

marioalvaro committed:
Create feature to check duplicates

Add new feature to check duplicate when adding a new task.
The feature has three modes:
* ON: This will check duplicates for the description and exact timings of the task
* TEXT: This will only check duplicates in the descriptions. If two events have the same description but different timings, it will also register as a duplicate
* OFF: No duplicate check will be executed

Having duplicates checker prevents cluttering of the list so that it is cleaner to look at.
But, the different modes are useful if we want to make recurring events (It is okay to have duplicates)

marioalvaro committed:
Add javadoc and checkstyle

Clean up several methods and write javadocs for every methods and class added as a result of adding duplicates method.

marioalvaro committed:
Merge pull request #17 from marioalvaro/branch-C-DetectDuplicates

Branch c detect duplicates

marioalvaro committed:
Add user guide

marioalvaro committed:
Fix User Guide

marioalvaro committed:
Fix User Guide #2
